lib/wallet.ts

```diff
@@ -0,0 +1,88 @@
+import { CryptoUtil } from './crypto-util.js';
+import { KeyEncryption, IEncryptedJSON } from './key-encryption.js';
+import PassPhraseGenerator from './pass-gen.js'
+import { Provider } from './provider.js';
+import { Signer } from './signer.js';
+import { SendMoney } from './transactions/send-money.js';
+
+
+
+export class Wallet extends Signer {
+    accountId: string;
+    accountRS: string;
+    provider: Provider | null;
+    constructor(publicKey: string, privKey: string, accountId: string, provider: Provider | null = null) {

```

Comment:
I would make the provider required of have a default provider since it seems like you need to typecast everywhere first of all and also you need to this is going to fail otherwise.

Reply:
This Wallet is exposed by the SDK and some people using it might not want a provider attached to their wallet.
Wallets can be used as an offline signer so no need to connect to a node (thus no need for a provider). 

Ethers js has the same approach here, to have the wallet independent of the provider. However in ethers concerning the provider, the wallet is immutable. When you call etherswallet.connect(provider) you will get a new wallet instance. In our GMD SDK you can just call gmdwallet.conntect(provider) and this null variable will be set.